apps/cow-tools/src/lib/priceCheckers.ts

```diff
@@ -56,23 +70,74 @@ export const priceCheckerInfoMapping = {
         label: "Allowed slippage (%)",
         inputType: "number",
         convertInput: (input: number) => BigInt(input * 100),
+        toExpectedOutCalculator: false,
       },
     ],
     name: PRICE_CHECKERS.SUSHI_SWAP,
+    hasExpectedOutCalculator: true,
+
     schema: sushiSwapSchema,
+    getSchema: getSushiSwapSchema,
   },
 } as const;
 
 export function encodePriceCheckerData(
```

Comment:
this became a bit harder to reason about, you could try modularizing / splitting the code to make it a bit easier to reason about again, e.g. (w/ the help of chatgpt):

```
export function encodePriceCheckerData(
  priceChecker: PRICE_CHECKERS,
  args: bigint[],
): `0x${string}` {
  const priceCheckerInfo = priceCheckerInfoMapping[priceChecker];

  validateArguments(priceCheckerInfo.arguments, args);
  
  if (!priceCheckerInfo.hasExpectedOutCalculator) {
    return encodeArguments(priceCheckerInfo.arguments, args);
  }

  return encodeWithExpectedOutCalculator(priceCheckerInfo.arguments, args);
}

function validateArguments(expectedArgs: PriceCheckerArgument[], args: bigint[]) {
  if (expectedArgs.length !== args.length || !args.length) {
    throw new Error(`Invalid number of arguments`);
  }
}

function encodeArguments(arguments: PriceCheckerArgument[], args: bigint[]): `0x${string}` {
  return encodeAbiParameters(arguments.map(arg => ({ name: arg.name, type: arg.type })), args);
}

function encodeWithExpectedOutCalculator(arguments: PriceCheckerArgument[], args: bigint[]): `0x${string}` {
  const firstExpectedOutArgIndex = arguments.findIndex(arg => arg.toExpectedOutCalculator);

  // If no expected out calculator arguments are found, encode all arguments as usual
  if (firstExpectedOutArgIndex === -1) {
    return encodeArguments(arguments, args);
  }

  // Encode arguments before the expected out calculator arguments
  const mainArgs = arguments.slice(0, firstExpectedOutArgIndex);
  const mainEncoded = encodeArguments(mainArgs, args.slice(0, firstExpectedOutArgIndex));

  // Encode arguments for the expected out calculator
  const expectedOutArgs = arguments.slice(firstExpectedOutArgIndex);
  const expectedOutEncoded = encodeArguments(expectedOutArgs, args.slice(firstExpectedOutArgIndex));

  // Combine the two encoded parts with a delimiter
  // This may need to be adjusted based on the specific ABI encoding requirements
  return `0x${mainEncoded.slice(2)}${expectedOutEncoded.slice(2)}`;
}
```

Reply:
Nice! I am doing this on the next PR!